client/hwlib/src/collectors/hardware_info.rs

```diff
@@ -78,10 +75,12 @@ impl TryFrom<(&SMBiosProcessorInformation<'_>, &Path)> for Processor {
 
     fn try_from(value: (&SMBiosProcessorInformation, &Path)) -> Result<Self> {
         let (processor_info, max_cpu_frequency_filepath) = value;
-        let cpu_id = CpuId::from_smbios(processor_info)?;
+        let cpu_id = processor_info
+            .processor_id()
+            .ok_or_else(|| anyhow!("processor ID not found"))?;
         let cpu_freq = CpuFrequency::from_k_hz_file(max_cpu_frequency_filepath)?.get_m_hz();
         Ok(Processor {
-            codename: cpu_id.codename().unwrap_or_else(|| "Unknown".to_string()),
+            identifier: *cpu_id,
```

Comment:
Not sure if it's something for this PR, but there's a minor inconsistency with the naming here as we have both `id` and `identifier` to refer to the same concept. If possible, it would be nice to stick to just one, although I don't know which would be easiest, or indeed whether this is too much of a hassle

Reply:
Updated in bb02da2